### DISCUSSION TITLE
Any plans on porting webpack to vite / swc?

### DISCUSSION THREAD

**Discussion by Aex12** [payloadcms/payload#606] [Feature Requests & Ideas]
I've been integrating Payload CMS with my NextJS project, and I'm very happy with it, however, since I started using it, building time has went up by a noticeable margin. I'm wondering if there are any plans on changing babel/webpack to a faster approach like Vite or SWC.

Thanks for opening up this awesome project!

**Comment by jmikrut** [comment]
Hey @Aex12 — good question!

We are totally interested in exploring this. We have some big plans to work through before we could get to this, and it'd come as a breaking change given our ability to customize / override Payload's Webpack config, but I could see it being very valuable. If you're interested in working up a PR, we'd be all about it — but if not, I'm sure we'll be evaluating this in the future.

**Comment by DanRibbens** [comment]
Aside from throwing out webpack completely, I found we can improve performance by turning on webpack filesystem cache. On the built-in `demo` project in the repo this is reducing webpack build times for me from 12 - 14s to 1 - 2s.

I'm running it on a new payload project with yarn link and I'm getting webpack build times consistently around 2s after the initial build.

https://github.com/payloadcms/payload/pull/621
